packages/salesforcedx-vscode-core/src/conflict/metadataCacheService.ts

```diff
@@ -57,33 +65,48 @@ export class MetadataCacheService {
    * @param componentPath A path referring to a project folder or an individual component resource
    * @param projectPath The base path of an sfdx project
    */
-  public initialize(componentPath: string, projectPath: string): void {
+  public initialize(
+    componentPath: string,
+    projectPath: string,
+    isManifest: boolean = false
+  ): void {
     this.componentPath = componentPath;
     this.projectPath = projectPath;
+    this.isManifest = isManifest;
   }
 
   /**
    * Load a metadata cache based on a project path that defines a set of components.
    *
-   * @param componentPath A path referring to a project folder or an individual component resource
+   * @param componentPath A path referring to a project folder, an individual component resource
+   * or a manifest file
    * @param projectPath The base path of an sfdx project
+   * @param isManifest Whether the componentPath references a manifest file
    * @returns MetadataCacheResult describing the project and cache folders
    */
   public async loadCache(
     componentPath: string,
-    projectPath: string
+    projectPath: string,
+    isManifest: boolean = false
   ): Promise<MetadataCacheResult | undefined> {
-    this.initialize(componentPath, projectPath);
+    this.initialize(componentPath, projectPath, isManifest);
     const components = await this.getSourceComponents();
     const operation = await this.createRetrieveOperation(components);
     const results = await operation.start();
     return this.processResults(results);
   }
 
-  public getSourceComponents(): ComponentSet {
-    return this.componentPath
-      ? (this.sourceComponents = ComponentSet.fromSource(this.componentPath))
-      : new ComponentSet();
+  public async getSourceComponents(): Promise<ComponentSet> {
+    if (this.componentPath && this.projectPath) {
+      this.sourceComponents = this.isManifest
+        ? await ComponentSet.fromManifest({
+            manifestPath: this.componentPath,
+            resolveSourcePaths: [this.projectPath]
```

Comment:
I think we need to make this a list of the project's package directories. If `projectPath` is the root of the project, it'll still work but will end up over-scanning irrelevant files like node_modules which could impact the performance